lib/api/dsl/methods.js

```diff
@@ -75,15 +74,15 @@ module.exports = methods = {
     async.each(Object.keys(filter[field]), function (fn, callback) {
       var
         value = filter[field][fn],
-        curriedFunctionName = '',
+        encodedFunctionName = '',
         result;
 
       if (not) {
-        curriedFunctionName += 'not';
+        encodedFunctionName += 'not';
       }
-      curriedFunctionName += 'range' + field + fn + value;
+      encodedFunctionName += 'range' + field + fn + value;
 
-      result = buildCurriedFunction(index, collection, field, fn, value, curriedFunctionName, roomId, not);
+      result = addToFiltersTree(index, collection, field, fn, value, encodedFunctionName, roomId, not);
```

Comment:
It's the range operator.

Here is a typical range filter:

```json
{
  "range": {
    "field": {
      "gt": 10,
      "lte": 123
    }
  }
}
```

The `fn` variable here refers to `gt` and `lte`. As it is not very clear in the code, I'll change this variable name to `rangeOperator`